iam_utility_node_group.tf

```diff
@@ -13,6 +13,8 @@ data "aws_iam_policy_document" "cluster_autoscaler_policy" {
       # https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html
       "ec2:DescribeInstanceTypes",
       "ec2:DescribeLaunchTemplateVersions",
+
+      "eks:DescribeNodegroup",
```

Comment:
Instead of adding this here can we separate it out and only allow for particular nodegroup resources? We're aiming for least privilege.

```
  statement {
    effect = "Allow"
    actions = [
      "eks:DescribeNodegroup"
    ]
    resources = [
      "arn:aws:eks:*:${local.aws_account_id}:nodegroup/redpanda-*"
    ]
  }
```